geograypher/meshes/meshes.py

```diff
@@ -162,19 +163,49 @@ def load_mesh(
             simplify_tol_meters=ROI_simplify_tol_meters,
         )
 
-        # Downsample mesh if needed
+        # Downsample mesh and transfer active scalars from original mesh to downsampled mesh
         if downsample_target != 1.0:
             # TODO try decimate_pro and compare quality and runtime
             # TODO see if there's a way to preserve the mesh colors
             # TODO also see this decimation algorithm: https://pyvista.github.io/fast-simplification/
             self.logger.info("Downsampling the mesh")
-            self.pyvista_mesh = self.pyvista_mesh.decimate(
+            # Have a temporary mesh so we can use the original mesh to transfer the active scalars to the downsampled one
+            downsampled_mesh_without_textures = self.pyvista_mesh.decimate(
                 target_reduction=(1 - downsample_target)
             )
+            self.pyvista_mesh = self.transfer_texture(downsampled_mesh_without_textures)
         self.logger.info("Extracting faces from mesh")
         # See here for format: https://github.com/pyvista/pyvista-support/issues/96
         self.faces = self.pyvista_mesh.faces.reshape((-1, 4))[:, 1:4].copy()
 
+    def transfer_texture(self, downsampled_mesh):
+        """Transfer texture from original mesh to a downsampled version using KDTree for nearest neighbor point searches
+
+        Args:
+            downsampled_mesh (pv.PolyData): The downsampled version of the original mesh
+
+        Returns:
+            pv.PolyData: The downsampled mesh with the transferred textures
+        """
+        # Only compute KDTree if active scalars are associated with points
+        if (self.pyvista_mesh.active_scalars_info.association == pv.FieldAssociation.POINT):
```

Comment:
I didn't catch this earlier, but I think this line could be a problem if there are no active scalars. This is completely possible, and something we saw on the first mesh we experimented with which I think was the example data provided with the repo. Can you find a mesh, or create one if needed, that has no scalars and test this? There should be an easy way to check if there are active scalars to fix this.

Reply:
I'm working on the implementation for transferring all the scalars and based on what I have so far I replaced that line with `if self.pyvista_mesh.point_data:`. I'm checking if there is any point-based scalar data, and if there is, transfer over all the point-based scalar data at the correct indices. So I think I won't be using the check above at all because I don't need to check the active scalars association type.